presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -334,6 +336,9 @@ protected void setup(Binder binder)
         // optimizers
         binder.bind(new TypeLiteral<List<PlanOptimizer>>() {}).toProvider(PlanOptimizersFactory.class).in(Scopes.SINGLETON);
 
+        // plan sanity checkers
+        binder.bind(new TypeLiteral<List<PlanSanityChecker>>() {}).toProvider(PlanSanityCheckersFactory.class).in(Scopes.SINGLETON);
```

Comment:
I don't think this abstraction is necessary, especially since they don't have dependencies on other classes (like metadata, etc). I would just handle all this within PlanSanityChecker (which could hold a list of checkers internally and iterate over them in the `validate()` call`